File: examples-sdk/swift/031-basic-mesh-routing/main.swift
// 031-basic-mesh-routing — Swift SDK equivalent
// Compile: vil compile --from swift --input 031-basic-mesh-routing/main.swift --release

let server = VilServer(name: "banking-transaction-mesh", port: 8080)
let teller = ServiceProcess(name: "teller")
teller.endpoint(method: "GET", path: "/ping", handler: "teller_ping")
teller.endpoint(method: "POST", path: "/submit", handler: "teller_submit")
server.service(teller)
let fraud_check = ServiceProcess(name: "fraud_check")
fraud_check.endpoint(method: "POST", path: "/analyze", handler: "fraud_process")
server.service(fraud_check)
let core_banking = ServiceProcess(name: "core_banking")
core_banking.endpoint(method: "POST", path: "/post", handler: "core_banking_post")
server.service(core_banking)
let notification = ServiceProcess(name: "notification")
notification.endpoint(method: "GET", path: "/send", handler: "notification_send")
server.service(notification)
server.compile()

File: examples-sdk/csharp/031-basic-mesh-routing/Main.cs
// 031-basic-mesh-routing — C# SDK equivalent
// Compile: vil compile --from csharp --input 031-basic-mesh-routing/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("banking-transaction-mesh", 8080);
var teller = new ServiceProcess("teller");
teller.Endpoint("GET", "/ping", "teller_ping");
teller.Endpoint("POST", "/submit", "teller_submit");
server.Service(teller);
var fraud_check = new ServiceProcess("fraud_check");
fraud_check.Endpoint("POST", "/analyze", "fraud_process");
server.Service(fraud_check);
var core_banking = new ServiceProcess("core_banking");
core_banking.Endpoint("POST", "/post", "core_banking_post");
server.Service(core_banking);
var notification = new ServiceProcess("notification");
notification.Endpoint("GET", "/send", "notification_send");
server.Service(notification);
server.Compile();
